Assets/MRTK/SDK/Inspectors/Toolbox/MixedRealityToolboxWindow.cs

```diff
@@ -0,0 +1,317 @@
+
+            /// <summary>
+            /// The prefab type categories represented by this collection.
+            /// </summary>
+            public ToolboxCategory[] Categories => categories;
+        }
+
+        /// <summary>
+        /// Represents a distinct categorization of prefabs for display in the toolbox window.
+        /// </summary>
+        [Serializable]
+        internal class ToolboxCategory
+        {
+            [SerializeField]
+            private string categoryName = string.Empty;
+
+            /// <summary>
+            /// The friendly name representing this category.
+            /// </summary>
+            public string CategoryName => categoryName;
+
+            [SerializeField]
+            private ToolboxItem[] items = null;
+
+            /// <summary>
+            /// The prefabs represented by this category.
+            /// </summary>
+            public ToolboxItem[] Items => items;
+        }
+
+        /// <summary>
+        /// Represents a prefab and additional data about the prefab for the toolbox.
+        /// </summary>
+        [Serializable]
+        internal class ToolboxItem
+        {
+            [SerializeField]
+            private string name = string.Empty;
+
+            /// <summary>
+            /// The friendly name of this prefab.
+            /// </summary>
+            public string Name => name;
+
+            [SerializeField]
+            private string docURL = string.Empty;
+
+            /// <summary>
+            /// A link to documentation about this prefab.
+            /// </summary>
+            public string DocURL => docURL;
+
+            [SerializeField]
+            private string assetGUID = string.Empty;
+
+            /// <summary>
+            /// The path from the AssetDatabase to the prefab.
+            /// </summary>
+            public string AssetPath => AssetDatabase.GUIDToAssetPath(assetGUID);
+
+            [SerializeField]
+            private string iconGUID = string.Empty;
+
+            /// <summary>
+            /// The path from the AssetDatabase to the icon representing this prefab in the toolbox UI.
+            /// </summary>
+            public string IconPath => AssetDatabase.GUIDToAssetPath(iconGUID);
+
+            /// <summary>
+            /// The actual prefab GameObject represented by this item.
+            /// </summary>
+            public GameObject Prefab { get; protected set; }
+
+            /// <summary>
+            /// The icon representing this prefab in the toolbox UI.
+            /// </summary>
+            public Texture Icon { get; protected set; }
+
+            /// <summary>
+            /// Initialize this item by loading the prefab and icon from the AssetDatabase.
+            /// </summary>
+            public void Init()
+            {
+                Prefab = AssetDatabase.LoadAssetAtPath<GameObject>(AssetPath);
+                Icon = AssetDatabase.LoadAssetAtPath<Texture>(IconPath);
```

Comment:
Proposal that I'm looking for feedback on (@cre8ivepark, @thalbern, @CDiaz-MS, @wiwei, @davidkline-ms):
Since this PR is adding ~2mb of images just for this tool, I'd prefer we somehow make those an optional import. Not sure if we move everything to the Tools package, or if we take this path, where we host our images somewhere (on the docs site?) and load them when the Toolbox is opened via the following code:

```suggestion
                Icon = AssetPreview.GetAssetPreview(Prefab);

                imageWebRequest = UnityWebRequestTexture.GetTexture(iconURL);
                imageWebRequest.SendWebRequest();
                EditorApplication.update += CheckForLoadedImage;
            }

            private void CheckForLoadedImage()
            {
                if (imageWebRequest.isDone)
                {
                    EditorApplication.update -= CheckForLoadedImage;
                    Icon = DownloadHandlerTexture.GetContent(imageWebRequest);
                }
```

The icon falls back to a Unity-generated prefab preview, which isn't as nice as the hand-created icons we'd be loading, but is something. That way, those without internet can still see an image.

Reply:
Sounds great!

Reply:
I like only loading the images (hosted on the docs site) if the toolbox is opened.  The unity generated images show the prefab turned around.  @cre8ivepark Do you know why the prefabs might show up turned around in the prefab preview?

Reply:
I'd vote for going with the tools package approach - I'm a bit wary of adding random calls to the internet in our toolkit, and if the concern is around the foundation package (totally valid), this really is a tool anyway (should be in the tools location).

Reply:
@wiwei It makes sense!

Reply:
I just pushed a change to move the Toolbox and its related resources (images and json data) from the SDK package to the Tools package.